x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/delete_custom_dashboard.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createRouteValidationFunction } from '@kbn/io-ts-utils';
+import { InfraDeleteCustomDashboardsRequestParamsRT } from '../../../common/http_api/custom_dashboards_api';
+import { KibanaFramework } from '../../lib/adapters/framework/kibana_framework_adapter';
+import { handleRouteErrors } from '../../utils/handle_route_errors';
+import { checkCustomDashboardsEnabled } from './lib/check_custom_dashboards_enabled';
+import { deleteCustomDashboard } from './lib/delete_custom_dashboard';
+import { findCustomDashboardByDashboardId } from './lib/find_custom_dashboard';
+
+export function initDeleteCustomDashboardRoute(framework: KibanaFramework) {
+  const validateParams = createRouteValidationFunction(InfraDeleteCustomDashboardsRequestParamsRT);
+
+  framework.registerRoute(
+    {
+      method: 'delete',
+      path: '/api/infra/{assetType}/custom-dashboards/{dashboardSavedObjectId}',
+      validate: {
+        params: validateParams,
+      },
+      options: {
+        access: 'internal',
+      },
+    },
+    handleRouteErrors(async (context, request, response) => {
+      const { savedObjectsClient, uiSettingsClient } = await context.infra;
+
+      await checkCustomDashboardsEnabled(uiSettingsClient);
+
+      const params = request.params;
+      const existingCustomDashboard = await findCustomDashboardByDashboardId(
+        params.dashboardSavedObjectId,
+        savedObjectsClient
+      );
+
+      await deleteCustomDashboard({
+        savedObjectsClient,
+        savedObjectId: existingCustomDashboard?.saved_objects[0]?.id ?? '',
```

Comment:
Do we want to delete even when existingCustomDashboard is undefined?

Reply:
Now that we pass the `id` (it's mandatory to use the endpoint) in the path we don't need to do that anymore (as we don't search for an `existingCustomDashboard`